course/templatetags/course.py

```diff
@@ -38,6 +39,32 @@ def group_select(context):
         'selected': selected,
     }
 
+@register.filter
+def parse_localization(entry):
+    """
+    Picks the currently selected language's value from
+    |lang:value|lang:value| -format text.
+    """
+    text = ""
+    if isinstance(entry, str):
+        text = entry
+    else:
+        text = str(entry) # if not a string, assume entry is a model
```

Comment:
`text  = entry if isinstance(entry, str) else str(entry)`. Though, `str(a_string_object)` would work too.

Other solution would be setting entry if it's not str:
`if not isinstance(entry, str): entry = str(entry)`